engine/input.cpp

```diff
@@ -28,9 +28,10 @@ void Input::Init(GLFWwindow* window)
         inputVal = InputState::NotPressed;
     }
 
-    glfwSetKeyCallback(window, Input::KeyCallback);
-    glfwSetCursorPosCallback(window, Input::MousePositionCallback);
-    glfwSetMouseButtonCallback(window, Input::MouseButtonCallback);
+    GLFWwindow* glfwWindow = ApplicationWindow::GetInstance().GetWindow();
```

Comment:
This definitely hides this dependency a little. If you ever try to unit test this, or change the implementation it will be messy to tear it apart again.